src/VisualStudio/Core/Def/TableDataSource/DiagnosticTableItem.cs

```diff
@@ -1,141 +0,0 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-using System;
-using System.Collections.Generic;
-using Microsoft.CodeAnalysis;
-using Microsoft.CodeAnalysis.Diagnostics;
-using Microsoft.CodeAnalysis.Text;
-using Roslyn.Utilities;
-
-namespace Microsoft.VisualStudio.LanguageServices.Implementation.TableDataSource;
-
-internal sealed class DiagnosticTableItem : TableItem
```

Comment:
after removing AbstractTableSnapshot, there are no more references to this type (or its base type).  so ths PR removes both of those as well.